app/celery/provider_tasks.py

```diff
@@ -74,14 +77,18 @@ def deliver_sms(
         )
         raise NotificationTechnicalFailureException from e
     except NonRetryableException as e:
-        # Max retries exceeded, celery raised exception
+        # Likely an opted out from pinpoint
```

Comment:
API #2092 is likely to modify the status_reason here to match the comment. When I saw the original comment I could not validate it had anything to do with retries.